oap-server/server-receiver-plugin/otel-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/otel/otlp/OpenTelemetryMetricRequestProcessor.java

```diff
@@ -135,8 +136,8 @@ private static Map<String, String> buildLabels(List<KeyValue> kvs) {
         return kvs
             .stream()
             .collect(toMap(
-                KeyValue::getKey,
-                it -> it.getValue().getStringValue()
+                it -> it.getKey().replaceAll("\\.", "_"),
```

Comment:
I think we can handle the label name as same as the metrics name in `PrometheusMetricConverter`,  and use a cache to reduce the costs. https://github.com/apache/skywalking/blob/86155bef811f748bf57fe14244ac2ecca916f8f1/oap-server/analyzer/meter-analyzer/src/main/java/org/apache/skywalking/oap/meter/analyzer/prometheus/PrometheusMetricConverter.java#L164
Note: this will impact some expressions in `envoy.yaml` and `envoy-svc-relation.yaml` .
https://github.com/apache/skywalking/blob/86155bef811f748bf57fe14244ac2ecca916f8f1/oap-server/server-starter/src/main/resources/envoy-metrics-rules/envoy.yaml#L63

Reply:
If the dot doesn't make the MAL engine have side effect, we don't have to change this, right? @yswdqz